Tasks/Gradle/gradletask.ts

```diff
@@ -124,6 +126,19 @@ function processCodeAnalysisResults(): Q.Promise<void> {
     return sqGradle.uploadSonarQubeBuildSummaryIfEnabled();
 }
 
+// Configure the maximum memory available to Gradle if the user hasn't done so, otherwise it may run ouf of heap space
+function configureMemoryLimits() {
+    if (!process.env['GRADLE_OPTS']) {
+
+        let defaultMavenOpts = '-Xmx4096m';
```

Comment:
4096 might be too much, 2048 might be a better value. When using SonarQube for msbuild we set it to 1024m and we had only 1 customer complain about out of memory exception.